### PR TITLE
Update Copier Template to 2.2.0

### LINKED ISSUE #736: Update copier template to include SLF001
Now the copier template includes [SLF001](https://docs.astral.sh/ruff/rules/private-member-access/) for production code (but not tests), we should make use of it as discussed here: https://github.com/DiamondLightSource/python-copier-template/issues/154

## Acceptance Criteria
* Copier template up-to-date (at least version 2.2.0)
* SLF001 enabled for production code

### PR DESCRIPTION
**Pull request by callumforrester**
Fixes #736 

### Instructions to reviewer on how to test:
1. Confirm tests pass
2. Try accessing a member with a ` _` in the `src` directory 

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)
- [ ] If changing the API for a pre-existing device, ensure that any beamlines using this device have updated their Bluesky plans accordingly
- [ ] Have the connection tests for the relevant beamline(s) been run via `dodal connect ${BEAMLINE}`

### CODE REVIEW

File: tests/common/test_coordination.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def test_group_uid(group: str):
 
 
 def test_type_checking_ignores_inject():
-    def example_function(x: Movable = inject("foo")) -> MsgGenerator:
+    def example_function(x: Movable = inject("foo")) -> MsgGenerator:  # noqa: B008
```
Comment (callumforrester) [inline tests/common/test_coordination.py:19]: [B008](https://docs.astral.sh/ruff/rules/function-call-in-default-argument/) is going to create some awkwardness for us, possibly even after #483 and associated work has been complete. I suggest raising another issue for dealing with it, unless that discussion has happened already.

File: pyproject.toml [hunk 2]
```diff
@@ -156,15 +157,18 @@ lint.extend-ignore = [
...
 ]
 lint.select = [
-    "B",    # flake8-bugbear - https://docs.astral.sh/ruff/rules/#flake8-bugbear-b
-    "C4",   # flake8-comprehensions - https://beta.ruff.rs/docs/rules/#flake8-comprehensions-c4
-    "E",    # pycodestyle errors - https://beta.ruff.rs/docs/rules/#error-e
-    "F",    # pyflakes rules - https://beta.ruff.rs/docs/rules/#pyflakes-f
-    "I",    # isort - https://docs.astral.sh/ruff/rules/#isort-i
-    "W",    # pycodestyle warnings - https://beta.ruff.rs/docs/rules/#warning-w
-    "UP",   # pyupgrade - https://docs.astral.sh/ruff/rules/#pyupgrade-up
-    "I001", # isort
```
Comment (callumforrester) [inline pyproject.toml:166]: I'm not sure why isort was left there by the previous template upgrade, I assume it is now fully replaced by ruff. Happy to put it back if I'm wrong though.

File: src/dodal/devices/areadetector/adutils.py [hunk 3]
```diff
@@ -31,7 +31,7 @@ def trigger(self):
 
         def _acq_done(*args, **kwargs):
             # TODO sort out if anything useful in here
-            self._status._finished()
+            self._status._finished()  # noqa: SLF001
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/dodal/devices/areadetector/adutils.py:34]: Am I right this will get removed in https://github.com/DiamondLightSource/dodal/issues/404?

**Review by DominicOram — APPROVED** [review]
Great, thank you!

File: src/dodal/devices/areadetector/adutils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (callumforrester) [reply src/dodal/devices/areadetector/adutils.py:34]: Not will, but _could_, yes!

**Comment by callumforrester** [comment]
@DominicOram thanks for rebasing, are you happy to merge with the lower coverage or do you want to wait for #404?

File: src/dodal/devices/areadetector/adutils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (DiamondJoseph) [reply src/dodal/devices/areadetector/adutils.py:34]: It has now been removed as part of #405, I didn't realise how much dead code was in that dir

**Pull request merged**